### PR TITLE
Fixes #37 - Handle RTF created with TX_RTF32

### LINKED ISSUE #37: suspicious behavior
Hey!

I was trying to decode an RFT file and ```rtf_to_text``` threw an error "Cannot decode a symbol \u0417 blah-blah".
I tried passing several different encodings to ```rtf_to_text``` but it didn't work.
Then I looked into the source code and found [this](https://github.com/joshy/striprtf/blob/febf3c05cc715a23dbead9dbed62354b0f79831d/striprtf/striprtf.py#L177) line. I have uncommented it and tried passing "blahblah" as the ```encoding``` expecting that it'd throw me a print, but it didn't.

Then I started to suspect that [this](https://github.com/joshy/striprtf/blob/febf3c05cc715a23dbead9dbed62354b0f79831d/striprtf/striprtf.py#L173) or [this](https://github.com/joshy/striprtf/blob/febf3c05cc715a23dbead9dbed62354b0f79831d/striprtf/striprtf.py#L178) line actually overrides the ```encoding``` argument that I pass to the function. After removing these two lines and setting the correct encoding everything went smoothly.

Idk what's going on there but you should probably look into it as this feels wrong :).

### PR DESCRIPTION
**Pull request by plessbd**
Fixes #37 

TX_RTF32 allows for UTF8 encoded characters within an cpg1252 encoded RTF

### CODE REVIEW

**Comment by joshy** [comment]
Thx a lot!

File: striprtf/striprtf.py [hunk 1]
```diff
@@ -190,17 +190,25 @@ def rtf_to_text(text, encoding="utf-8", errors="strict"):
                     c = int(arg)
                     if c < 0:
                         c += 0x10000
-                    out = out + chr(c).encode(encoding, errors)
+                    try:
+                        chr1 = chr(c).encode(encoding, errors).decode(encoding).encode('utf-8', errors)
+                    # because some flavors of rtf seem to allow utf8 inside of cp1252
+                    except UnicodeEncodeError as e:
+                        chr1 = chr(c).encode('utf-8', errors)
```
Comment (stevengj) [inline striprtf/striprtf.py:197]: I think you should just execute this line unconditionally.

What is the point of first trying to encode it through `encoding`?  It's a Unicode character — just encode it directly into UTF-8.

File: striprtf/striprtf.py [hunk 2]
```diff
@@ -149,21 +153,18 @@ def rtf_to_text(text, encoding="utf-8", errors="strict"):
             curskip = 0
             if char == "~":
                 if not ignorable:
-                    out = out + b"\xA0"  # NBSP
+                    out += b"\xA0".decode('utf-8')
```
Comment (stevengj) [inline striprtf/striprtf.py:156]: This seems incorrect.  `decode` assumes that  `\xA0` is UTF-8, which it is *not*.    It is the Unicode codepoint for [U+00A0](https://www.fileformat.info/info/unicode/char/00a0/index.htm) (non-breaking space).

I think you want
```py
out += chr(0xa0).encode('utf-8') # NBSP
```
here, or simply hard-code the UTF-8 encoding:
```py
out += b'\xc2\xa0' # NBSP in UTF-8
```

File: striprtf/striprtf.py [hunk 3]
```diff
@@ -119,17 +119,21 @@ def rtf_to_text(text, encoding="utf-8", errors="strict"):
...
     ucskip = 1  # Number of ASCII characters to skip after a unicode character.
     curskip = 0  # Number of ASCII characters left to skip
-    out = b''  # Output buffer.
+    hexes = None
+    out = ''
 
     for match in PATTERN.finditer(text):
         word, arg, _hex, char, brace, tchar = match.groups()
+        if hexes and not _hex:
+            out += bytes.fromhex(hexes).decode(encoding=encoding).encode('utf8').decode()
```
Comment (stevengj) [inline striprtf/striprtf.py:135]: Why is the final `.decode()` needed here?  Don't you just want
```py
out += bytes.fromhex(hexes).decode(encoding=encoding).encode('utf8')
```
?

File: striprtf/striprtf.py [hunk 4]
```diff
@@ -190,17 +190,25 @@ def rtf_to_text(text, encoding="utf-8", errors="strict"):
                     c = int(arg)
                     if c < 0:
                         c += 0x10000
-                    out = out + chr(c).encode(encoding, errors)
+                    try:
+                        chr1 = chr(c).encode(encoding, errors).decode(encoding).encode('utf-8', errors)
+                    # because some flavors of rtf seem to allow utf8 inside of cp1252
+                    except UnicodeEncodeError as e:
+                        chr1 = chr(c).encode('utf-8', errors)
+                    out += chr1.decode('utf-8')
```
Comment (stevengj) [inline striprtf/striprtf.py:198]: Why not simply:
```py
out += chr(c)
```
Why have the UTF-8 step at all?

File: striprtf/striprtf.py [hunk 5]
```diff
@@ -149,21 +153,18 @@ def rtf_to_text(text, encoding="utf-8", errors="strict"):
...
             elif char == "*":
                 ignorable = True
             elif char == "\n":
                 if not ignorable:
-                    out = out + b"\x0A"  # LF
+                    out += b"\x0A".decode('utf-8')
             elif char == "\r":
                 if not ignorable:
-                    out = out + b"\x0D"  # CR
+                    out += b"\x0D".decode('utf-8')
```
[the remark this answers is not in the document]
Comment (stevengj) [reply striprtf/striprtf.py:167]: i.e. just do
```py
out += char
```

File: striprtf/striprtf.py (hunk 1 above)
Comment (stevengj) [reply striprtf/striprtf.py:197]: better yet, skip the encoding step entirely as commented below

File: striprtf/striprtf.py [hunk 6]
```diff
@@ -190,17 +190,25 @@ def rtf_to_text(text, encoding="utf-8", errors="strict"):
...
+                    hexes = _hex
+                else:
+                    hexes += _hex
         elif tchar:
             if curskip > 0:
                 curskip -= 1
             elif not ignorable:
-                out = out + tchar.encode(encoding, errors)
-    return out.decode(encoding, errors)
+                out += tchar.encode('utf-8').decode()
```
Comment (stevengj) [inline striprtf/striprtf.py:213]: Again, just
```py
out += tchar
```
seems like it should suffice here — encode + decode is a no-op.

**Comment by joshy** [comment]
Oh wow, I will review the changes. Thanks @stevengj

**Pull request merged**